Cargo.toml

```diff
@@ -12,4 +12,9 @@ keywords = ["interval", "tree", "bound", "exclusive", "difference"]
 categories = ["algorithms", "data-structures"]
 
 [dependencies]
-rand = "0.8"
\ No newline at end of file
+rand = "0.8"
+serde = { version = "1.0.145", optional = true , features = ["derive"] }
```

Comment:
Is there a reason to select version "1.0.145" specifically? I think I would prefer to just specify "1.0", that is we don't limit to a specific patch number.